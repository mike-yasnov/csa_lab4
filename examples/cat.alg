// Программа cat - печатает данные, поданные через ввод

function main() {
    var ch = 1;  // Инициализируем не нулём
    while (ch != 0) {
        ch = read();
        if (ch != 0) {
            print(ch);
        }
    }
}

main(); 