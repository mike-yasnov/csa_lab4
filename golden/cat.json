{
  "return_code": 0,
  "stdout": "Загрузка программы: /tmp/tempfile/program.bin\nЗагружено 24 инструкций\nЗагрузка данных: /tmp/tempfile/program_data.bin\nЗагружено 4 байт данных\nЗапуск выполнения (максимум 1000000 тактов)...\n\n=== РЕЗУЛЬТАТЫ ВЫПОЛНЕНИЯ ===\nСостояние: halted\nВыполнено инструкций: 230\nЗатрачено тактов: 697\nФинальный PC: 23\n\nВЫВОД ПРОГРАММЫ:\nHello\nWorld\n\n\nПрограмма завершена успешно.\n",
  "stderr": "",
  "exec_log": "Cycle 000002: PC=0000 40(000016) Stack[0] TOS=empty\nCycle 000006: PC=0016 43(000001) Stack[0] TOS=empty\nCycle 000008: PC=0001 00(000001) Stack[0] TOS=empty\nCycle 000010: PC=0002 00(000000) Stack[1] TOS=1\nCycle 000014: PC=0003 51(000000) Stack[2] TOS=0\nCycle 000016: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000020: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000022: PC=0006 00(000000) Stack[1] TOS=1\nCycle 000025: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000028: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000033: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000035: PC=000A 00(000000) Stack[1] TOS=72\nCycle 000039: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000041: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000045: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000047: PC=000E 00(000000) Stack[1] TOS=72\nCycle 000050: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000053: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000055: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000059: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000064: PC=0013 61(000002) Stack[1] TOS=72\nCycle 000066: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000068: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000072: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000074: PC=0006 00(000000) Stack[1] TOS=72\nCycle 000077: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000080: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000085: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000087: PC=000A 00(000000) Stack[1] TOS=101\nCycle 000091: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000093: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000097: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000099: PC=000E 00(000000) Stack[1] TOS=101\nCycle 000102: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000105: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000107: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000111: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000116: PC=0013 61(000002) Stack[1] TOS=101\nCycle 000118: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000120: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000124: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000126: PC=0006 00(000000) Stack[1] TOS=101\nCycle 000129: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000132: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000137: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000139: PC=000A 00(000000) Stack[1] TOS=108\nCycle 000143: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000145: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000149: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000151: PC=000E 00(000000) Stack[1] TOS=108\nCycle 000154: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000157: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000159: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000163: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000168: PC=0013 61(000002) Stack[1] TOS=108\nCycle 000170: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000172: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000176: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000178: PC=0006 00(000000) Stack[1] TOS=108\nCycle 000181: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000184: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000189: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000191: PC=000A 00(000000) Stack[1] TOS=108\nCycle 000195: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000197: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000201: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000203: PC=000E 00(000000) Stack[1] TOS=108\nCycle 000206: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000209: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000211: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000215: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000220: PC=0013 61(000002) Stack[1] TOS=108\nCycle 000222: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000224: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000228: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000230: PC=0006 00(000000) Stack[1] TOS=108\nCycle 000233: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000236: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000241: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000243: PC=000A 00(000000) Stack[1] TOS=111\nCycle 000247: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000249: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000253: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000255: PC=000E 00(000000) Stack[1] TOS=111\nCycle 000258: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000261: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000263: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000267: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000272: PC=0013 61(000002) Stack[1] TOS=111\nCycle 000274: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000276: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000280: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000282: PC=0006 00(000000) Stack[1] TOS=111\nCycle 000285: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000288: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000293: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000295: PC=000A 00(000000) Stack[1] TOS=10\nCycle 000299: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000301: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000305: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000307: PC=000E 00(000000) Stack[1] TOS=10\nCycle 000310: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000313: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000315: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000319: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000324: PC=0013 61(000002) Stack[1] TOS=10\nCycle 000326: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000328: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000332: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000334: PC=0006 00(000000) Stack[1] TOS=10\nCycle 000337: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000340: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000345: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000347: PC=000A 00(000000) Stack[1] TOS=87\nCycle 000351: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000353: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000357: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000359: PC=000E 00(000000) Stack[1] TOS=87\nCycle 000362: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000365: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000367: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000371: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000376: PC=0013 61(000002) Stack[1] TOS=87\nCycle 000378: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000380: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000384: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000386: PC=0006 00(000000) Stack[1] TOS=87\nCycle 000389: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000392: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000397: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000399: PC=000A 00(000000) Stack[1] TOS=111\nCycle 000403: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000405: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000409: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000411: PC=000E 00(000000) Stack[1] TOS=111\nCycle 000414: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000417: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000419: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000423: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000428: PC=0013 61(000002) Stack[1] TOS=111\nCycle 000430: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000432: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000436: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000438: PC=0006 00(000000) Stack[1] TOS=111\nCycle 000441: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000444: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000449: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000451: PC=000A 00(000000) Stack[1] TOS=114\nCycle 000455: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000457: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000461: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000463: PC=000E 00(000000) Stack[1] TOS=114\nCycle 000466: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000469: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000471: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000475: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000480: PC=0013 61(000002) Stack[1] TOS=114\nCycle 000482: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000484: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000488: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000490: PC=0006 00(000000) Stack[1] TOS=114\nCycle 000493: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000496: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000501: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000503: PC=000A 00(000000) Stack[1] TOS=108\nCycle 000507: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000509: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000513: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000515: PC=000E 00(000000) Stack[1] TOS=108\nCycle 000518: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000521: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000523: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000527: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000532: PC=0013 61(000002) Stack[1] TOS=108\nCycle 000534: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000536: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000540: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000542: PC=0006 00(000000) Stack[1] TOS=108\nCycle 000545: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000548: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000553: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000555: PC=000A 00(000000) Stack[1] TOS=100\nCycle 000559: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000561: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000565: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000567: PC=000E 00(000000) Stack[1] TOS=100\nCycle 000570: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000573: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000575: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000579: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000584: PC=0013 61(000002) Stack[1] TOS=100\nCycle 000586: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000588: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000592: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000594: PC=0006 00(000000) Stack[1] TOS=100\nCycle 000597: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000600: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000605: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000607: PC=000A 00(000000) Stack[1] TOS=10\nCycle 000611: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000613: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000617: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000619: PC=000E 00(000000) Stack[1] TOS=10\nCycle 000622: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000625: PC=0010 41(000014) Stack[1] TOS=1\nCycle 000627: PC=0011 00(000000) Stack[0] TOS=empty\nCycle 000631: PC=0012 50(000000) Stack[1] TOS=0\nCycle 000636: PC=0013 61(000002) Stack[1] TOS=10\nCycle 000638: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000640: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000644: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000646: PC=0006 00(000000) Stack[1] TOS=10\nCycle 000649: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000652: PC=0008 41(000015) Stack[1] TOS=1\nCycle 000657: PC=0009 60(000000) Stack[0] TOS=empty\nCycle 000659: PC=000A 00(000000) Stack[1] TOS=0\nCycle 000663: PC=000B 51(000000) Stack[2] TOS=0\nCycle 000665: PC=000C 00(000000) Stack[0] TOS=empty\nCycle 000669: PC=000D 50(000000) Stack[1] TOS=0\nCycle 000671: PC=000E 00(000000) Stack[1] TOS=0\nCycle 000674: PC=000F 31(000000) Stack[2] TOS=0\nCycle 000677: PC=0010 41(000014) Stack[1] TOS=0\nCycle 000679: PC=0014 40(000004) Stack[0] TOS=empty\nCycle 000681: PC=0004 00(000000) Stack[0] TOS=empty\nCycle 000685: PC=0005 50(000000) Stack[1] TOS=0\nCycle 000687: PC=0006 00(000000) Stack[1] TOS=0\nCycle 000690: PC=0007 31(000000) Stack[2] TOS=0\nCycle 000693: PC=0008 41(000015) Stack[1] TOS=0\nCycle 000696: PC=0015 44(000000) Stack[0] TOS=empty\nCycle 000697: PC=0017 70(000000) Stack[0] TOS=empty",
  "debug_listing": "INSTRUCTION MEMORY:\nAddress - Hex Code - Mnemonic\n----------------------------------------\n0000 - 40160000 - JMP 22\n0002 - 00010000 - PUSH 1\n0004 - 00000000 - PUSH\n0006 - 51000000 - STORE\n0007 - 00000000 - PUSH\n0009 - 50000000 - LOAD\n000A - 00000000 - PUSH\n000C - 31000000 - NE\n000D - 41150000 - JZ 21\n000F - 60000000 - IN\n0011 - 00000000 - PUSH\n0013 - 51000000 - STORE\n0014 - 00000000 - PUSH\n0016 - 50000000 - LOAD\n0017 - 00000000 - PUSH\n0019 - 31000000 - NE\n001A - 41140000 - JZ 20\n001C - 00000000 - PUSH\n001E - 50000000 - LOAD\n001F - 61020000 - OUT 2\n0021 - 40040000 - JMP 4\n0023 - 44000000 - RET\n0024 - 43010000 - CALL 1\n0026 - 70000000 - HALT\n\nDATA MEMORY:\nAddress - Hex Dump - ASCII\n----------------------------------------\n0000 - 00 00 00 00                                      - ....\n"
}